### ISSUE TITLE
Intellisense does not correctly handle [ContentProperty] on DataTemplateSelector

### ISSUE THREAD

**Issue by mattleibow** [dotnet/maui#12017]
### Description

When creating a custom `DataTemplateSelector`, the `[ContentProperty]` attribute does not appear to be respected by the IDE and gives an error:

> XLS0504: Property 'DataTemplateContent' does not support values of type 'DynamicDataTemplate'.

This breaks hot reload, but is actually valid XAML (I think?) and compiles and runs just fine.

```cs
[ContentProperty(nameof(Templates))]
public class DynamicTemplateSelector : DataTemplateSelector
{
    protected override DataTemplate OnSelectTemplate(object item, BindableObject container);

    public ObservableCollection<DynamicDataTemplate> Templates { get; } = new();
}
```

```xaml
<CollectionView ItemsSource="{Binding Items}">
  <CollectionView.ItemTemplate>
    <local:DynamicTemplateSelector>
      <local:DynamicDataTemplate>
        <DataTemplate>
          <Label Text="{Binding .}" />
        </DataTemplate>
      </local:DynamicDataTemplate>
    </local:DynamicTemplateSelector>
  </CollectionView.ItemTemplate>
</CollectionView>
```

### Steps to Reproduce

1. Clone repro (https://github.com/mattleibow/maui-bug-repros/tree/repro-xaml-dts)
2. Open the `MainPage.xaml` in the IDE
3. Observe the intellisense error
4. Run / F5
5. Observe the app works fine

### Link to public reproduction project repository

https://github.com/mattleibow/maui-bug-repros/tree/repro-xaml-dts

### Version with bug

7.0 (current)

### Last version that worked well

Unknown/Other

### Affected platforms

iOS, Android, Windows, macOS, Other (Tizen, Linux, etc. not supported by Microsoft directly)

### Affected platform versions

All versions

### Did you find any workaround?

Add the explicit property element.

From:

```xaml
<local:DynamicTemplateSelector>
    ...
</local:DynamicTemplateSelector>
```

To:

```xaml
<local:DynamicTemplateSelector>
    <local:DynamicTemplateSelector.Templates>
    ...
    </local:DynamicTemplateSelector.Templates>
</local:DynamicTemplateSelector>
```

**Comment by jfversluis** [comment]
This seems like something @StephaneDelcroix could know?

**Comment by StephaneDelcroix** [comment]
@mgoertz-msft is the definitive source of knowledge on this topic. but it might indicates we miss a converter in one of our Design assembly

**Comment by Eilon** [comment]
@mattleibow - can you get in touch with @marcpopMSFT via email/Teams to see if this should stay here, or move to AzDO? If it should move to AzDO please use the VS Feedback tool.

**Comment by XamlTest** [comment]
Verified this on Visual Studio Enterprise 17.7.0 Preview 1.0. Repro on Windows 11, Android 13.0-API33 and  iOS 16.4 with below Project:
[MauiApp1.zip](https://github.com/dotnet/maui/files/11718794/MauiApp1.zip)

**Comment by marcpopMSFT** [comment]
> @mattleibow - can you get in touch with @marcpopMSFT via email/Teams to see if this should stay here, or move to AzDO? If it should move to AzDO please use the VS Feedback tool.

Didn't see this the first time but no idea where to route this as it's not the SDK. Sounds like IntelliSense so maybe Roslyn. CC @jaredpar

**Comment by jaredpar** [comment]
This sounds like XAML to me but that's admittedly a bit of a guess.

**Comment by mgoertz-msft** [comment]
Correct, that looks like a XAML IntelliSense issue. @mattleibow You may want to either directly open a bug or create a VS feedback ticket for any further public engagement.

**Comment by kubaflo** [comment]
Thanks for the report. The remaining problem belongs to Visual Studio, the debugger/editor, or the .NET MAUI VS Code extension rather than the MAUI runtime.

Please report it through the [Visual Studio feedback tool](https://learn.microsoft.com/visualstudio/ide/how-to-report-a-problem-with-visual-studio) or the [.NET MAUI VS Code extension repository](https://github.com/microsoft/vscode-dotnettools/issues), as appropriate. I am closing this MAUI issue so the runtime backlog only tracks work actionable in this repository; if you file the tooling report, please link it here.

**Comment by kubaflo** [comment]
Reopening — this issue should not have been closed.

It carries `s/verified` (a maintainer or partner reproduced it), `s/triaged` (it already went through triage), `partner/ide` (a partner team is tracking it). Those labels mean the report was already confirmed as a real product defect, so it is not eligible for backlog cleanup closure. Closure here requires specific fix evidence (a merged PR that closes it, or a `fixed-in-*` label), and there was none.

Sorry for the noise, and thanks to everyone who flagged it.

> Disclosure: this comment was generated by AI.
